Microsoft.Toolkit.Uwp.Services/Services/MicrosoftTranslator/AzureAuthToken.cs

```diff
@@ -0,0 +1,133 @@
+﻿// ******************************************************************
+// Copyright (c) Microsoft. All rights reserved.
+// This code is licensed under the MIT License (MIT).
+// THE CODE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+// IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+// TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH
+// THE CODE OR THE USE OR OTHER DEALINGS IN THE CODE.
+// ******************************************************************
+
+using System;
+using System.Net.Http;
+using System.Threading.Tasks;
+
+namespace Microsoft.Toolkit.Uwp.Services.MicrosoftTranslator
+{
+    /// <summary>
+    /// Client to call Cognitive Services Azure Auth Token service in order to get an access token.
+    /// </summary>
+    internal class AzureAuthToken
+    {
+        /// <summary>
+        /// Name of header used to pass the subscription key to the token service
+        /// </summary>
+        private const string OcpApimSubscriptionKeyHeader = "Ocp-Apim-Subscription-Key";
+
+        /// <summary>
+        /// URL of the token service
+        /// </summary>
+        private static readonly Uri ServiceUrl = new Uri("https://api.cognitive.microsoft.com/sts/v1.0/issueToken");
+
+        /// <summary>
+        /// After obtaining a valid token, this class will cache it for this duration.
+        /// Use a duration of 5 minutes, which is less than the actual token lifetime of 10 minutes.
+        /// </summary>
+        private static readonly TimeSpan TokenCacheDuration = new TimeSpan(0, 5, 0);
+
+        private readonly HttpClient _client;
+
+        private string _storedTokenValue = string.Empty;
+        private DateTime _storedTokenTime = DateTime.MinValue;
+        private string _subscriptionKey;
+
+        /// <summary>
+        /// Gets or sets the Service Subscription Key.
+        /// </summary>
+        public string SubscriptionKey
+        {
+            get
+            {
+                return _subscriptionKey;
+            }
+
+            set
+            {
+                if (_subscriptionKey != value)
+                {
+                    // If the subscription key is changed, the token is no longer valid.
+                    _subscriptionKey = value;
+                    _storedTokenValue = string.Empty;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureAuthToken"/> class, that is used to obtain access token
+        /// </summary>
+        /// <param name="key">Subscription key to use to get an authentication token.</param>
+        public AzureAuthToken(string key)
+        {
+            SubscriptionKey = key;
+            _client = new HttpClient();
```

Comment:
The community created an optimized httpHelper  class in the toolkit for making http request, I see it's better to use it